dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -44,7 +44,7 @@ internal KernelPromptTemplate(PromptTemplateConfig promptConfig, bool allowUnsaf
         AddMissingInputVariables(this._blocks, promptConfig);
 
         this._allowUnsafeContent = allowUnsafeContent || promptConfig.AllowUnsafeContent;
-        this._safeBlocks = promptConfig.InputVariables.Where(iv => allowUnsafeContent || iv.AllowUnsafeContent).Select(iv => iv.Name).ToList();
+        this._safeBlocks = new HashSet<string>(promptConfig.InputVariables.Where(iv => allowUnsafeContent || iv.AllowUnsafeContent).Select(iv => iv.Name).ToList());
```

Comment:
No need to do a ToList here since HashSet already takes an IEnumerable as a parameter.  That should avoid another traversal list traversal.